### PR TITLE
feat!: Updating versions of CUDA, cuDNN, TensorRT and PyTorch

### PR DESCRIPTION
**Pull request by narendasan**
# Description

Updates dependency library versions:

- PyTorch version has been bumped to 1.8.0
- Default CUDA version is CUDA 11.1
- TensorRT version is TensorRT 7.2.3.4
- cuDNN version is now cuDNN 8.1

## Type of change

Please delete options that are not relevant and/or add your own.

- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [x] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by narendasan** [comment]
@peri044 can you take a look. make sure to delete and redownload all the modules from `hub.py`.

File: tests/py/test_api.py [hunk 1]
```diff
@@ -79,7 +79,6 @@ def test_is_colored_output_on(self):
 def test_suite():
     suite = unittest.TestSuite()
     suite.addTest(TestCompile.parametrize(TestCompile, model=models.resnet18(pretrained=True)))
-    suite.addTest(TestCompile.parametrize(TestCompile, model=models.resnet50(pretrained=True)))
```
Comment (peri044) [inline tests/py/test_api.py:82]: removing this since one variant of resnet (resnet 18) already exists ?

**Review by peri044 — COMMENTED** [review]
LGTM. Had some minor questions. Tested it on my local and it runs fine.

File: tests/py/test_api.py (hunk 1 above)
Comment (narendasan) [reply tests/py/test_api.py:82]: yeah

File: core/lowering/passes/linear_to_addmm.cpp [hunk 2]
```diff
@@ -7,29 +7,31 @@ namespace core {
...
             %res = aten::linear(%input, %weight, %bias)
             return (%res))IR";
 
+  std::string fused_linear = R"IR(
+        graph(%input, %weight_t, %bias):
+            %1: int = prim::Constant[value=1]()
+            %weight = aten::t(%weight_t)
+            %mm: Tensor = aten::matmul(%input, %weight)
+            %b_f: Tensor = trt::const(%bias)
+            %out: Tensor = aten::add_(%b_f, %mm, %1)
```
[the remark this answers is not in the document]
Comment (narendasan) [reply core/lowering/passes/linear_to_addmm.cpp:28]: just changes less ops, we probably should be using all functional operations but the functionalization pass of pytorch doesnt seem to do it for us

[the remark this answers is not in the document]
Comment (narendasan) [reply core/lowering/passes/linear_to_addmm.cpp:28]: I dont think it should be a blocker for this PR though, we can change it with auto fallback if needed

[the remark this answers is not in the document]
Comment (peri044) [reply core/lowering/passes/linear_to_addmm.cpp:28]: Yes it shouldn't. We can revisit this later.

**Pull request merged**